packages/commonwealth/server/routes/writeUserSetting.ts

```diff
@@ -22,38 +21,39 @@ const writeUserSetting = async (
   res: Response,
   next: NextFunction,
 ) => {
-  const { key, value } = req.body;
+  const {
+    disable_rich_text,
+    promotional_emails_enabled,
+    email_notification_interval,
+  } = req.body;
 
   if (!req.user) {
     return next(new AppError(Errors.InvalidUser));
   }
-  if (!key || !value) {
-    return next(new AppError(Errors.NoKeyValue));
-  }
 
-  if (key === 'disableRichText' && value === 'true') {
-    req.user.disableRichText = true;
-    await req.user.save();
-  } else if (key === 'disableRichText' && value === 'false') {
-    req.user.disableRichText = false;
-    await req.user.save();
-  } else if (
-    key === 'updateEmailInterval' &&
-    VALID_DIGEST_INTERVALS.indexOf(value) !== -1
-  ) {
-    req.user.emailNotificationInterval = value;
-    await req.user.save();
-  } else if (
-    key === 'promotional_emails_enabled' &&
-    ['true', 'false'].includes(value)
+  if (
+    !(
+      typeof disable_rich_text !== 'boolean' ||
+      typeof promotional_emails_enabled !== 'boolean' ||
+      email_notification_interval
```

Comment:
This logic seems incorrect. If `disable_rich_text !== 'boolean'` is true then the conditional evaluates to false and an InvalidSetting error is not returned.